com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,225 @@
+import com.chartboost.heliumsdk.ad.HeliumFullscreenAd
+import com.chartboost.heliumsdk.ad.HeliumRewardedAd
+import com.unity3d.player.UnityPlayer
+
+class HeliumUnityAdWrapper(private val ad: HeliumAd) {
+    // This is the container for the banner.  We need a relative layout to position the banner in one of the 7
+    // possible positions.  HeliumBannerAd is just a FrameLayout.
+    private val activity: Activity = UnityPlayer.currentActivity
+    private var bannerLayout: RelativeLayout? = null
+    private var startedLoad = false
+
+    private fun asBanner(): HeliumBannerAd? {
+        return ad as HeliumBannerAd?
+    }
+
+    private fun asFullScreen(): HeliumFullscreenAd? {
+        return ad as HeliumFullscreenAd?
+    }
+
+    fun load() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            ad.load()
+            startedLoad = true
+        }
+    }
+
+    fun load(screenLocation: Int) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            createBannerLayout(screenLocation)
+            load()
+        }
+    }
+
+    fun show() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            if (ad is HeliumFullscreenAd)
+                asFullScreen()?.show()
+        }
+    }
+
+    fun setKeyword(keyword: String, value: String): Boolean {
+        return ad.keywords.set(keyword, value)
+    }
+
+    fun removeKeyword(keyword: String): String {
+        return ad.keywords.remove(keyword) ?: ""
+    }
+
+    fun setCustomData(customData: String) {
+        if (ad is HeliumRewardedAd) {
+            ad.customData = customData
+        } else {
+            Log.d(TAG,"custom data can only be set on a rewarded ad")
+        }
+    }
+
+    fun setBannerVisibility(isVisible: Boolean) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            if (ad is HeliumBannerAd && bannerLayout != null) {
+                val visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
+                bannerLayout?.visibility = visibility
+                asBanner()?.visibility = visibility
+            }
+        }
+    }
+
+    fun clearLoaded(): Boolean {
+        if (ad is HeliumFullscreenAd)
+            return asFullScreen()?.clearLoaded() ?: false
+        if (ad is HeliumBannerAd) {
+            HeliumUnityBridge.runTaskOnUiThread { asBanner()?.clearAd() }
+            return true
+        }
+        return false
+    }
+
+    fun readyToShow(): Boolean {
+        return if (!startedLoad) false else try {
+            when (ad) {
+                is HeliumFullscreenAd -> asFullScreen()?.readyToShow() ?: false
+                is HeliumBannerAd -> {
+                    Log.w(TAG, "This should never be called, banners do not have readyToShow")
+                    false
+                }
+                else -> false
+            }
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - ${ex.message}")
+            false
+        }
```

Comment:
The `return if (!startedLoad) false else` can be simplified:

```kt
    return startedLoad && try {
        when (ad) {
            is HeliumFullscreenAd -> asFullScreen()?.readyToShow() ?: false
            is HeliumBannerAd -> {
                Log.w(TAG, "This should never be called, banners do not have readyToShow")
                false
            }
            else -> false
        }
    } catch (ex: Exception) {
        Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - ${ex.message}")
        false
    }
```

Reply:
Yup, auto convertion issue, should have caught it myself.